vgen/src/main/java/io/gapi/vgen/CodeGenerator.java

```diff
@@ -40,36 +32,40 @@
  */
 public class CodeGenerator {
 
-  private final GapicLanguageProvider provider;
+  private final GapicLanguageProvider<ProtoElement> provider;
 
-  public CodeGenerator(GapicLanguageProvider provider) {
+  public CodeGenerator(GapicLanguageProvider<ProtoElement> provider) {
     this.provider = Preconditions.checkNotNull(provider);
   }
 
-  public static CodeGenerator create(ConfigProto configProto, Model model) {
-    return new CodeGenerator(GeneratorBuilderUtil.createLanguageProvider(configProto, model));
+  public static CodeGenerator create(ConfigProto configProto, TemplateProto template, Model model) {
+    return new CodeGenerator(
+        GeneratorBuilderUtil.createLanguageProvider(configProto, template, model)); // FIXME
   }
 
   /**
    * Generates code for the model. Returns a map from service interface to code for the service.
    * Returns null if generation failed.
    */
   @Nullable
-  public Map<Interface, GeneratedResult> generate(SnippetDescriptor snippetDescriptor) {
+  public Map<ProtoElement, GeneratedResult> generate(SnippetDescriptor snippetDescriptor) {
+    ProtoElementViewing<ProtoElement> viewing = provider.getViewing();
+    Iterable<ProtoElement> elts = viewing.getIterable(provider.getModel());
+
     // Establish required stage for generation.
     provider.getModel().establishStage(Merged.KEY);
     if (provider.getModel().getErrorCount() > 0) {
       return null;
     }
 
     // Run the generator for each service.
-    ImmutableMap.Builder<Interface, GeneratedResult> generated = ImmutableMap.builder();
-    for (Interface iface : provider.getModel().getSymbolTable().getInterfaces()) {
-      if (!iface.isReachable()) {
+    ImmutableMap.Builder<ProtoElement, GeneratedResult> generated = ImmutableMap.builder();
+    for (ProtoElement elt : elts) {
```

Comment:
Generally in Java, words will be fully spelled out, so use element/elements instead of elt/elts